app/controllers/devise/registrations_controller.rb

```diff
@@ -83,9 +83,10 @@ def cancel
   protected
 
   def update_needs_confirmation?(resource, previous)
-    resource.respond_to?(:pending_reconfirmation?) &&
-      resource.pending_reconfirmation? &&
-      previous != resource.unconfirmed_email
+    resource.respond_to?(:can_send_reconfirmation?) &&
```

Comment:
I don't think it's OK to use `can_send_reconfirmation?` over `pending_reconfirmation?` inside the controller/view, as we can let the the controller do its work without the model sending the new email. Can you please revert these changes?